core/adapters/json.go

```diff
@@ -0,0 +1,29 @@
+package adapters
+
+import (
+	"encoding/json"
+	"math/big"
+)
+
+// JSONBigFloat accepts both string and float JSON values.
+type JSONBigFloat big.Float
```

Comment:
We have a bunch of util objects like this in the `utils` package this should probably live along side.